### PR TITLE
chore: Makes use of DS components in ThemeControls

### LINKED ISSUE #18714: [Task]: Canvas property pane components should be built using Design system components
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

As of now, we have custom components used in canvas property pane. These should be migrated to make use of design system components

### PR DESCRIPTION
**Pull request by aswathkk**
## Description

After doing an exploration, I found that ThemeShadowControl and ThemeBorderRadiusControl were not using the components from the Design System. Edited those components to make use of DS components.
Also removed some unwanted DOM elements

Fixes #18714 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
Tested manually by making sure the shadow selection & border radius selection works properly in Theming property pane and widget property pane

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/components/propertyControls/BoxShadowOptionsControl.tsx [hunk 1]
```diff
@@ -18,24 +18,18 @@ export interface BoxShadowOptionsControlProps extends ControlProps {
...
-      content={
-        <div>
-          <div>{optionKey}</div>
-        </div>
-      }
+      content={optionKey}
       key={optionKey}
       openOnTargetFocus={false}
     >
-      <button tabIndex={-1}>
```
Comment (jsartisan) [inline app/client/src/components/propertyControls/BoxShadowOptionsControl.tsx:29]: why are we removing tabIndex = -1?

File: app/client/src/pages/Editor/ThemePropertyPane/controls/ThemeShadowControl.tsx [hunk 2]
```diff
@@ -37,31 +38,36 @@ function ThemeBoxShadowControl(props: ThemeBoxShadowControlProps) {
...
+            <CloseLineIcon className="text-gray-700" />
+          )}
+        </div>
+      </TooltipComponent>
+    ),
+    value: optionKey,
+  }));
+
   return (
-    <div className="grid grid-flow-col gap-2 auto-cols-max">
```
Comment (jsartisan) [inline app/client/src/pages/Editor/ThemePropertyPane/controls/ThemeShadowControl.tsx:41]: I am assuming the layouting with grid will be going in Button tab component right?

File: app/client/src/components/propertyControls/BoxShadowOptionsControl.tsx (hunk 1 above)
Comment (aswathkk) [reply app/client/src/components/propertyControls/BoxShadowOptionsControl.tsx:29]: We don't need this DOM element at all. It's working without that <button> tag.

File: app/client/src/pages/Editor/ThemePropertyPane/controls/ThemeShadowControl.tsx (hunk 2 above)
Comment (aswathkk) [reply app/client/src/pages/Editor/ThemePropertyPane/controls/ThemeShadowControl.tsx:41]: Yes. The grid was not required here. It is done using flex in the ButtonTab component.

**Review by jsartisan — APPROVED** [review]

**Review by jsartisan — APPROVED** [review]

**Pull request merged**